lib/modem_key_mgmt/modem_key_mgmt.c

```diff
@@ -17,6 +17,8 @@
 
 LOG_MODULE_REGISTER(modem_key_mgmt, CONFIG_MODEM_KEY_MGMT_LOG_LEVEL);
 
+/* Protect the shared scratch_buf with a mutex. */
+K_MUTEX_DEFINE(key_mgmt_mutex);
```

Comment:
This can be `static`, also, a very minor comment, I would use a binary semaphore here, since I don't expect this API to be reentrant.

Reply:
There doesn't seem to be much difference  in cost

With mutex:
```
FLASH:      291868 B
```

With semaphore:
```
FLASH:      291872 B
```

So I kept it as a semaphore as it was 4 bytes smaller :D

Reply:
That's fine.